modules/data.atmosphere/R/met2model.module.R

```diff
@@ -0,0 +1,43 @@
+.met2model.module <- function(ready.id, model, con, stage, host, dir, met, str_ns, site, start_date, end_date, browndog, new.site, overwrite=FALSE) {
+  # Determine output format name and mimetype
+  model_info <- db.query(paste0("SELECT f.name, f.id, mt.type_string from modeltypes as m", 
+                                " join modeltypes_formats as mf on m.id = mf.modeltype_id",
+                                " join formats as f on mf.format_id = f.id",
+                                " join mimetypes as mt on f.mimetype_id = mt.id",
+                                " where m.name = '", model, "' AND mf.tag='met'"), con)
+  
+  if (model_info[1] == "CF Meteorology"){
+    stage$met2model=FALSE
+  }
+  
+  if(stage$met2model == TRUE){
+    logger.info("Begin Model Specific Conversion")
+    
+    formatname <- model_info[1]
+    mimetype   <- model_info[3]
+    
+    print("Convert to model format")
+    
+    input.id  <- ready.id$input.id[1]
+    outfolder <- ifelse(host$name == "localhost", 
+                        file.path(dir, paste0(met, "_", model,"_site_", str_ns)),
+                        file.path(host$dbfiles, paste0(met,"_", model,"_site_", str_ns)))
+    
+    pkg       <- paste0("PEcAn.",model)
+    fcn       <- paste0("met2model.",model)
+    lst       <- site.lst(site, con)
+    
+    model.id  <- convert.input(input.id, outfolder, formatname, mimetype, site.id=site$id,
+                               start_date, end_date, pkg, fcn, con=con, host=host, browndog, write=TRUE, 
+                               lst=lst, lat=new.site$lat, lon=new.site$lon, overwrite=overwrite)
+    
+  } else {
+    model.id = ready.id
+    
+    if("CRUNCEP" %in% met){outfolder <- file.path(dir,paste0(met,"_site_",str_ns))}
+    if("GFDL" %in% met){outfolder <- file.path(dir,paste0(met,"_site_",str_ns))}
```

Comment:
These hacks need to be collapsed (they're identical) and in general (throughout met.process) replaced with a registry tag identifying all cases where met is directly extracted (the number of products in this category is growing rapidly)